authorize_error.go

```diff
@@ -34,47 +34,49 @@ func (f *Fosite) WriteAuthorizeError(rw http.ResponseWriter, ar AuthorizeRequest
 	rw.Header().Set("Pragma", "no-cache")
 
 	rfcerr := ErrorToRFC6749Error(err)
+	if !f.SendDebugMessagesToClients {
+		rfcerr = rfcerr.Sanitize()
+	}
+
 	if !ar.IsRedirectURIValid() {
-		if !f.SendDebugMessagesToClients {
-			rfcerr.Debug = ""
-		}
+		rw.Header().Set("Content-Type", "application/json;charset=UTF-8")
 
 		js, err := json.MarshalIndent(rfcerr, "", "\t")
 		if err != nil {
-			http.Error(rw, err.Error(), http.StatusInternalServerError)
```

Comment:
This should probably always be JSON, not that sometimes it is just some plain test string.

Reply:
Correct!